### PR TITLE
WebView Js.nullable instead of option

### LINKED ISSUE #151: WebView Js.nullable instead of option
It seems to me that in WebView.rei, we should change `option(t)` to `Js.nullable(t)`
```reason
~onNavigationStateChange: {
  .
  "url": option(string),
  "title": option(string),
  "loading": option(bool),
  "canGoBack": option(bool),
  "canGoForward": option(bool)
} => unit
```
to
```reason
~onNavigationStateChange: {
  .
  "url": Js.nullable(string),
  "title": Js.nullable(string),
  "loading": Js.nullable(bool),
  "canGoBack": Js.nullable(bool),
  "canGoForward": Js.nullable(bool)
} => unit
```

Indeed no conversion is done in `WebView.re`

### PR DESCRIPTION
**Pull request by maarekj**
Fix #151

### CODE REVIEW

**Comment by maarekj** [comment]
Today, with option, it break.
Then changing to `Js.nullable` will not break compatibility because it never worked.

**Comment by wokalski** [comment]
@maarekj You're right. Nevertheless we (quite) consistently expose `option` type and use `Nullalble` behind the scenes.

**Comment by maarekj** [comment]
Obviously, it's best with the option type.
More difficult to implement, but better.

**Comment by wokalski** [comment]
@maarekj let me know if you're planning to implement the suggestion.

**Comment by maarekj** [comment]
@wokalski Done

File: src/components/webView.re [hunk 1]
```diff
@@ -119,12 +119,62 @@ let make =
             "injectedJavaScript": fromOption(injectedJavaScript),
             "mediaPlaybackRequiresUserAction":
               fromOption(mediaPlaybackRequiresUserAction),
-            "onError": fromOption(onError),
-            "onLoad": fromOption(onLoad),
-            "onLoadEnd": fromOption(onLoadEnd),
-            "onLoadStart": fromOption(onLoadStart),
+            "onError": fromOption(UtilsRN.option_map((callback) => {
```
Comment (wokalski) [inline src/components/webView.re]: This is repeated multiple times. Could you factor it out into a utility function?

**Comment by maarekj** [comment]
It seems that the `onNavigationStateChange` callback takes an additional parameter compared to the others.
It is `jsEvaluationValue`(as you can see on [this link](https://medium.com/capriza-engineering/communicating-between-react-native-and-the-webview-ac14b8b8b91a))

**Comment by wokalski** [comment]
@maarekj it makes little difference. Feel free to leave that one as is then.

**Comment by maarekj** [comment]
ping @wokalski

**Comment by wokalski** [comment]
@maarekj I'm slightly confused. Does `onNavigationStateChange` take the other argument?

**Comment by maarekj** [comment]
No, I understood that it should not be added.
If we add it, we can no longer use the newly created function `optionalCallbackToJs` and the new type `callbackParams`

**Comment by wokalski** [comment]
Hey, I'm closing this until there's further progress. The needed change is very small.

**Pull request closed without merging**